src/components/Courses/Courses.jsx

```diff
@@ -1,45 +1,113 @@
-import React, { useState, useMemo } from 'react';
-import './Courses.css';
+import React, { useState, useEffect } from 'react';
+import { useNavigate } from 'react-router-dom';
 import PropTypes from 'prop-types';
 import SearchBar from './components/SearchBar/SearchBar';
 import CourseCard from './components/CourseCard/CourseCard';
 import Button from '../../common/Button/Button';
+import CourseInfo from '../CourseInfo/CourseInfo';
+import { api_endpoints, storageKeys, paths } from '../../constants';
+import './Courses.css';
 
-const Courses = ({ courses, authors, onAddCourseClick }) => {
+const Courses = ({ onAddCourseClick }) => {
+  const [courses, setCourses] = useState([]);
+  const [authors, setAuthors] = useState([]);
   const [searchTerm, setSearchTerm] = useState('');
+  const [selectedCourse, setSelectedCourse] = useState(null);
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    const token = localStorage.getItem(storageKeys.userToken);
+    if (token) {
+      navigate(paths.courses);
+    }
+    fetchCourses();
+    fetchAuthors();
+  }, [navigate]);
 
-  const handleSearch = (term) => {
+  const fetchCourses = async (query = '', type = '') => {
+    try {
+      let response;
+      if (query!=='' && type!=='') {
+        response = await fetch(`${api_endpoints.filter}${type}=${query}`);
+      }
+      else{
+        response = await fetch(`${api_endpoints.courses}`);
+      }
+      
+      if (response.ok) {
+        const data = await response.json();
+        setCourses(data.result);
+      } else {
+        console.error('Failed to fetch courses:', response.status);
+      }
+    } catch (error) {
+      console.error('Error fetching courses:', error);
+    }
+  };
+
+  const fetchAuthors = async () => {
+    try {
+      const response = await fetch(api_endpoints.authors);
+      if (response.ok) {
+        const data = await response.json();
+        setAuthors(data.result);
+      } else {
+        console.error('Failed to fetch authors:', response.status);
+      }
+    } catch (error) {
+      console.error('Error fetching authors:', error);
+    }
+  };
+  const handleSearch = async (term) => {
     setSearchTerm(term);
+    await fetchCourses(term, 'title');
+    if (courses.length === 0) {
```

Comment:
As I understand you're checking the state variable, however there might be an issue with this approach. React's setState is not fully synchronous, so after you run `await fetchCourses` you might not have the correct data for `courses` at that point.

So, it's netter for `fetchCourses` to return the data.result as well. And then you can check for length of this returned value (if this is needed)